src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2717,12 +2718,11 @@ private static void LinkBlocks(BasicBlockBuilder previous, IOperation condition,
         /// Returns converted test expression.
         /// Caller is responsible for spilling the stack and pushing a stack frame before calling this helper.
         /// </summary>
-        private IOperation NullCheckAndConvertCoalesceValue(ICoalesceOperation operation, BasicBlockBuilder whenNull)
+        private IOperation NullCheckAndConvertCoalesceValue(IOperation operationValue, IConvertibleConversion coalesceConversion, ITypeSymbol coalesceType, BasicBlockBuilder whenNull)
```

Comment:
I am also not sure if it makes sense to reuse this function for CoalesceAssignment at all. It might be better to extract its first part (that deals with branching) into a separate helper and avoid dealing with faking conversions 

---
In reply to: [209388084](https://github.com/dotnet/roslyn/pull/29063#discussion_r209388084) [](ancestors = 209388084)